src/app/subnets/views/VLANDetails/ConfigureDHCP/ConfigureDHCP.tsx

```diff
@@ -101,24 +101,32 @@ const ConfigureDHCP = ({ closeForm, id }: Props): JSX.Element | null => {
         is: (val: string) => isId(val),
         then: Yup.string().required("Start IP address is required"),
       }),
-      subnet: Yup.string().test(
-        "hasNoIPs",
-        "Selected subnet has not available IPs",
-        (subnetId, context) => {
-          if (isId(subnetId)) {
-            const subnet = subnets.find(
-              (subnet) => subnet.id === Number(subnetId)
-            );
-            if (subnet?.statistics.num_available === 0) {
-              return context.createError({
-                message: "This subnet has no available IP addresses.",
-                path: "subnet",
-              });
+      subnet: Yup.string()
+        .when(["enableDHCP", "dhcpType"], {
+          is: (
+            dhcpEnabled: boolean,
+            dhcpType: DHCPType.CONTROLLERS | DHCPType.RELAY
+          ) => dhcpEnabled && DHCPType.CONTROLLERS === dhcpType,
+          then: Yup.string().required("Subnet is required"),
```

Comment:
When reconfiguring DHCP it looks like it might need to know if a subnet has already been set. If you go to https://maas-ui-4246.demos.haus/MAAS/r/vlan/5004/ and click "Configure DHCP" and then change the radio to "Provide DHCP from rack controller(s)" and select a controller the form is still disabled (which I suspect is because of this change).